### DISCUSSION TITLE
Clarification on Authentication with Docker Compose and .dockerignore

### DISCUSSION THREAD

**Discussion by AntoineMahe** [lllyasviel/Fooocus#3354] [Q&A] [answered]
**Hi everyone,** 👋

I'm currently facing some confusion regarding authentication in a Docker Compose setup, specifically around the `auth.json` file and the use of `.dockerignore`. I'd appreciate your insights and guidance on this matter.

**Current Understanding:**

1. The `auth.json` file appears to be needed inside the container for authentication purposes.
2. However, my `.dockerignore` file includes `/auth.json`, which means the default behavior is that this file won't be present in the container build context.

**Questions:**

- Is there a preferred way to handle this situation?
- Should I simply edit the `.dockerfile` to exclude `auth.json` from being ignored? Or is there another best practice to follow here?

Please feel free to link relevant documentation that  may have escape my search.

Thanks

**Comment by mashb1t** [comment, marked as answer]
There is a best practice, to mount the file manually and not to build it into the image as passwords may be changed during runtime and images could be pushed somewhere unsafe.
Therefore it is best to not delete it from the .dockerignore, but it should be manually added to the docker-compose.yml mounts.
